File: project_euler/Problem086/Problem086.py
import math

class Problem086:
    def solution(self) -> None:
        target, count, result = 1000000, 0, 1
        while count < target:
            result += 1
            for i in range(2, result):
                root = math.sqrt(result * result + i * i)
                if abs(math.floor(root + 0.5) - root) < 0.00000001:
                    count += i >> 1
            for i in range((result << 1) - 1, result - 1, -1):
                root = math.sqrt(result * result + i * i)
                if abs(math.floor(root + 0.5) - root) < 0.00000001:
                    count += result - ((i + 1) >> 1) + 1
        print(result)

File: project_euler/Problem086/Problem086.js
function Problem086() {
    let target = 1000000
    let count = 0
    let result = 1
    while (count < target) {
        ++result
        for (let i = 3; i < result; ++i) {
            let root = Math.sqrt(result * result + i * i)
            if (Math.abs(Math.floor(root + 0.5) - root) < 0.00000001) {
                count += i >>> 1
            }
        }
        for (let i = (result << 1) - 1; i >= result; --i) {
            let root = Math.sqrt(result * result + i * i)
            if (Math.abs(Math.floor(root + 0.5) - root) < 0.00000001) {
                count += result - ((i + 1) >>> 1) + 1
            }
        }
    }
    console.log(result)
}

module.exports = {
    Problem086 : Problem086
};
